spec/models/manageiq/providers/hawkular/inventory/parser/middleware_manager_spec.rb

```diff
@@ -97,79 +97,149 @@ def inventory_object_data(inventory_object)
+                                    :hawkular_metric_id => 'm2')
+        elsif feed == "f2"
+          metrics << OpenStruct.new(:path               => '/t;hawkular/f;2/r;s1/r;d1/m;m3',
+                                    :hawkular_metric_id => 'm3')
+          metrics << OpenStruct.new(:path               => '/t;hawkular/f;2/r;s1/r;d2/m;m4',
+                                    :hawkular_metric_id => 'm4')
+          metrics << OpenStruct.new(:path               => '/t;hawkular/f;2/r;s1/r;d3/m;m2',
+                                    :hawkular_metric_id => 'm2')
+          metrics << OpenStruct.new(:path               => '/t;hawkular/f;2/r;s1/r;d4/m;m5',
+                                    :hawkular_metric_id => 'm5')
+        end
+
+        metrics
+      end
+      allow(parser.collector).to receive(:raw_availability_data) do |ids|
+        avail_data = []
+        ids.each do |id|
+          case id
+          when "m1"
+            avail_data << OpenStruct.new(:id => id, :data => [{'timestamp' => 1, 'value' => 'up'}])
+          when "m2"
+            avail_data << OpenStruct.new(:id => id, :data => [{'timestamp' => 2, 'value' => 'down'}])
+          when "m3"
+            avail_data << OpenStruct.new(:id => id, :data => [{'timestamp' => 3, 'value' => 'something_else'}])
+          when "m4"
+            avail_data << OpenStruct.new(:id => id, :data => [])
+          end
+        end
+        avail_data
+      end
+
+      # Test deployment statuses are correctly assigned.
+      parser.fetch_deployment_availabilities(%w(f1 f2))
+      expect(deployments[0].status).to eq('Enabled')
+      expect(deployments[1].status).to eq('Disabled')
+      expect(deployments[2].status).to eq('Unknown')
+      expect(deployments[3].status).to eq('Unknown')
+      expect(deployments[4].status).to eq('Disabled')
+      expect(deployments[5].status).to eq('Unknown') # Missing metric
     end
-    it 'handles missing metrics' do
-      resources_by_metric_id = {
-        'resource_id_1' => [{}],
-        'resource_id_2' => [{}],
-        'resource_id_3' => [{}],
-        'resource_id_4' => [{}],
-      }
-      availabilities = [
-        {
-          'id'   => 'resource_id_1',
-          'data' => [{ 'value' => 'up' }]
-        }
-      ]
-
-      parsed_resources_with_availability = {
-        'resource_id_1' => [{
-          :status => 'Enabled'
-        }],
-        'resource_id_2' => [{}],
-        'resource_id_3' => [{}],
-        'resource_id_4' => [{}]
-      }
-      expect(parser.parse_availability(availabilities, resources_by_metric_id))
-        .to eq(parsed_resources_with_availability)
+  end
+
+  describe 'fetch_server_availabilities' do
+    it 'resolves availabilities correctly' do
+      parser = described_class.new
+
+      # Define a mocked persister.
+      # It has two feeds:
+      # - Feed 1 has one server with id: s1
+      # - Feed 2 has two servers ids: s1, s2
+      parser.persister = double("Mocked persister")
```

Comment:
Never mind... I was thinking in creating a mock-class and do something like:
`persister.parser` = MockedPersister.new
and I couldn't see how to use `instance_double`. But you were referring only to the behaviour in your other comment. That makes sense :)